src/error.ts

```diff
@@ -84,36 +84,44 @@ export class CertificateError extends Error {
 						host: url.hostname,
 						rejectUnauthorized: false,
 					},
-					() => {
+					async () => {
 						const x509 = socket.getPeerX509Certificate();
 						socket.destroy();
 						if (!x509) {
 							throw new Error("no peer certificate");
 						}
 
-						// We use node-forge for two reasons:
-						// 1. Node/Electron only provide extended key usage.
-						// 2. Electron's checkIssued() will fail because it suffers from same
-						//    the key usage bug that we are trying to work around here in the
-						//    first place.
-						const cert = forge.pki.certificateFromPem(x509.toString());
-						if (!cert.issued(cert)) {
+						/**
+						 * We use "@peculiar/x509" for two reasons:
+						 * 1. Node's x509 returns an undefined `keyUsage` in Electron environments.
```

Comment:
Oh wait does Node return undefined `keyUsage` only inside Electron?  Outside Electron it does actually exist?